Sources/PrimerSDK/Classes/User Interface/ApplePayPresentationManager.swift

```diff
@@ -77,20 +79,48 @@ class ApplePayPresentationManager: ApplePayPresenting, LogReporter {
         return request
     }
 
-    func shippingContactFields(applePayOptions: PrimerApplePayOptions?) -> Set<PKContactField> {
-        guard applePayOptions?.shippingOptions?.isCaptureShippingAddressEnabled == true else {
-            return []
-        }
+    func mapContactFieldsFromOptions(applePayOptions: PrimerApplePayOptions?) -> (mappedShippingContactFields: Set<PKContactField>, mappedBillingContactFields: Set<PKContactField>) {
 
-        var fields: Set<PKContactField> = [.postalAddress]
+        var requiredShippingContactFields = Set<PKContactField>()
+        var requiredBillingContactFields = Set<PKContactField>()
 
-        if let additionalFields = applePayOptions?.shippingOptions?.additionalShippingContactFields {
-            additionalFields.forEach {
-                fields.insert($0.toPKContact())
+        // Map required shipping contact fields
+        if let shippingContactFields = applePayOptions?.shippingOptions?.shippingContactFields, !shippingContactFields.isEmpty {
+            shippingContactFields.forEach {
+                requiredShippingContactFields.insert($0.toPKContact())
             }
         }
 
-        return fields
+        // Map required billing contact fields
+        if let billingContactFields = applePayOptions?.billingOptions?.requiredBillingContactFields {
+            billingContactFields.forEach {
+                requiredBillingContactFields.insert($0.toPKContact())
+            }
+        }
+
+        // Handle deprecated `isCaptureBillingAddressEnabled`
+        if requiredBillingContactFields.isEmpty, applePayOptions?.isCaptureBillingAddressEnabled == true {
+            requiredBillingContactFields.insert(.postalAddress)
+            requiredBillingContactFields.insert(.name)
+        }
+
+        // Move phone and email from billing to shipping if existing
+        let phoneField = PKContactField.phoneNumber
+        let emailField = PKContactField.emailAddress
+
+        if requiredBillingContactFields.contains(phoneField), !requiredShippingContactFields.contains(phoneField) {
+            requiredShippingContactFields.insert(phoneField)
+        }
+
+        if requiredBillingContactFields.contains(emailField), !requiredShippingContactFields.contains(emailField) {
+            requiredShippingContactFields.insert(emailField)
+        }
+
+        // Remove phone and email from billing fields
+        requiredBillingContactFields.remove(phoneField)
+        requiredBillingContactFields.remove(emailField)
+
+        return (requiredShippingContactFields, requiredBillingContactFields)
```

Comment:
@euljr please check this and compare it to WEB code 🙏

Reply:
✅ LGTM!